### ISSUE TITLE
Warning given about use of posterior_linpred when using fitted_draws with a count regression in rstanarm

### ISSUE THREAD

**Issue by kaybenleroll** [mjskay/tidybayes#278]
I am using rstanarm to fit a count regression model (I will use poisson for this reprex)

```
library(tidyverse)
library(purrr)
library(rstanarm)
library(tidybayes)

set.seed(42)


data_tbl <- tribble(
  ~category, ~mean_freq, ~n_row,
        "A",       0.07,   3000,
        "B",       0.03,   5000,
        "C",       0.05,   2000,
    ) %>%
  mutate(n_count = map2(n_row, mean_freq, ~ rpois(n = .x, lambda = .y))) %>%
  select(category, n_count) %>%
  unnest(n_count)

newdata_tbl <- sample(c("A", "B", "C"), size = 100, replace = TRUE) %>%
  enframe(name = NULL, value = "category")


regress_stanglm <- stan_glm(
    n_count ~ category,
    data            = data_tbl,
    family          = poisson(),
    chains          = 4,
    iter            = 1000,
    prior_intercept = normal(location = -4, scale = 1),
    prior           = normal(location =  0, scale = 1),
    QR              = TRUE
    )


new_linpred_tbl <- newdata_tbl %>%
  add_linpred_draws(
    model = regress_stanglm,
    value = "mean_freq"
    )
```

By running this, the `add_linpred_draws()` omits the following warning. I don't think it effects the results, but I figure I should report it and I'm happy to try to help out fixing it if I get some pointers.

```
Instead of posterior_linpred(..., transform=TRUE) please call posterior_epred(), which provides equivalent functionality.
```

**Comment by mjskay** [comment]
Ah yup, thanks! I think I knew this was coming. As you say, the results should still be correct.

I think ultimately I want to rethink the naming (and possibly defaults) of add_linpred_draws, and create a different function (probably add_epred_draws) to mirror the new name of posterior_epred --- which is actually a better name, reflecting that it is draws from E(y). 

This is going to be a delicate operation because the default of add_linpred_draws is currently scale = "response" when perhaps it should be scale = "linear". That might be a problem for another day, though.

The other related change would be to deprecate fitted_draws / add_fitted_draws, since those names are ambiguous compared to linpred / epred

Basically I see the changes as something like:

- [ ] make fitted_draws / add_fitted_draws be aliases for linpred_draws / add_linpred_draws instead of the other way around (which is the current approach)
- [ ] probably add a warning for [add_]linpred_draws on rstanarm objects with scale = "response" telling people to use scale = "linear" or add_epred_draws depending on what they actually want
- [ ] make epred_draws for rstanarm / brms, patterned after linpred_draws. Possibly this will necessitate some refactoring to avoid code duplication. Would need to think about the default column name of the output, as we probably could make a more specific one than the generic `.value` used for linpred. Possibly `.epred` or `.E` or `.mean` or something like that.

If you're interested in helping, I'll gladly accept help! The most relevant file for this is https://github.com/mjskay/tidybayes/blob/master/R/fitted_draws.R

**Comment by kaybenleroll** [comment]
I'm certainly interested - let me have a look at this over the next few days and I'll see about getting started.

**Comment by mjskay** [comment]
thanks!

**Comment by mjskay** [comment]
okay this should be fixed on the dev branch now via #285 --- [add_]fitted_draws is now deprecated in favor of using whichever of [add_]epred_draws or [add_]linpred_draws is appropriate, and those two functions should not give you warnings in their default configurations.
